zk_prover/src/merkle_sum_tree/mod.rs

```diff
@@ -7,11 +7,15 @@ pub mod utils;
 use halo2_proofs::halo2curves::bn256::Fr as Fp;
 
 #[derive(Clone, Debug)]
-pub struct MerkleProof<const N_ASSETS: usize, const N_BYTES: usize> {
```

Comment:
To clarify the meaning of the numbers adjacent to `N_ASSETS`, I would like to suggest an update as follows:

```Rust
#[derive(Clone, Debug)]
pub struct MerkleProof<const N_ASSETS: usize, const N_BYTES: usize>
where
    [usize; 1 + N_ASSETS]: Sized, // Adding 1 for the username
    [usize; N_ASSETS + 2]: Sized, // Adding 2 for two hashes
{
    pub entry: Entry<N_ASSETS>,
    pub root: Node<N_ASSETS>,
    pub sibling_leaf_node_hash_preimage: [Fp; 1 + N_ASSETS],
    pub sibling_middle_node_hash_preimages: Vec<[Fp; N_ASSETS + 2]>,
    pub path_indices: Vec<Fp>,
}
```

This suggestion makes the code more intuitive, and it should help newcomers more easily understand the hash inputs.

Reply:
Applied in [94404ed](https://github.com/summa-dev/summa-solvency/pull/188/commits/94404ed931227f9fa51e6a4be17ffe7ee4e564ae)